convert.go

```diff
@@ -115,11 +115,13 @@ func FromString(s string) (Date, error) {
 // FromTime validates that a `time.Time{}` contains a date and converts it to a
 // `Date{}`.
 func FromTime(t time.Time) (Date, error) {
+	hasLocationInfo := !(t.Location() == time.UTC || t.Location().String() == "")
```

Comment:
Given the naming, wouldn't it make sense for this to be

```go
hasLocationInfo := t.Location() == time.UTC || t.Location().String() == ""
```

and then use `!hasLocationInfo` on line 124?

Reply:
and then the naming doesn't make sense.
I can make it inline and remove the variable. Shall I do that?

Reply:
Oh I was thinking about it wrong, I get it now!

Reply:
I think the failing test showcases an issue with this approach

```
-timestamp contains more than just date information; 2022-01-31T00:00:00-05:00
+<nil>
```

In particular, the [test case][1] ends up resulting in a `time.FixedZone("", -5*60*60)` which isn't a "zero offset" even though it has a name equal to `""`. Note that `time.Location{}.String()` just [returns][2] the name and does not mention the offset.

----

Do you have an example value deserialized by `lib/pq`? Could you `fmt.Printf("%#v\n", tz)` the value and peek at the internal fields?

[1]: https://github.com/hardfinhq/go-date/blob/249e6fc1d29c720a14fc186c917c4323d704cf37/convert_test.go#L93-L94
[2]: https://github.com/golang/go/blob/go1.22.2/src/time/zoneinfo.go#L103-L105

Reply:
Actually `t.Zone()` may do the trick for us here